dockerfiles/executor_image/Dockerfile

```diff
@@ -51,3 +67,51 @@ RUN ARCH=$(dpkg --print-architecture) && \
     rm docker-credential-gcr.tar.gz && \
     docker-credential-gcr configure-docker
+
+RUN ARCH=$(case $(dpkg --print-architecture) in \
+        amd64) echo 'x86_64' ;; \
+        arm64) echo 'aarch64' ;; \
+    esac) && \
+    ROOTLESSKIT_VERSION="v2.1.0" && \
+    SHASUM=$(case $ARCH in \
+        x86_64) echo '1e4609a3d0ca67c4fc99fe9121870cf5ca46a438f43b151e9b54456d166f4b10' ;; \
+        aarch64) echo 'b3a9b14b66f54351a2b9f8c0917062bfeab2dc89e9e7cb15f4cf0684a4a7281f' ;; \
+    esac) && \
+    curl -o "rootlesskit.tar.gz" -fsSL "https://github.com/rootless-containers/rootlesskit/releases/download/${ROOTLESSKIT_VERSION}/rootlesskit-${ARCH}.tar.gz" && \
+    echo "$SHASUM rootlesskit.tar.gz" | sha256sum -c && \
+    tar -xz -f rootlesskit.tar.gz --to-stdout rootlesskit > /usr/bin/rootlesskit && \
+    tar -xz -f rootlesskit.tar.gz --to-stdout rootlesskit-docker-proxy > /usr/bin/rootlesskit-docker-proxy && \
+    chmod +x /usr/bin/rootlessctl /usr/bin/rootlesskit /usr/bin/rootlesskit-docker-proxy && \
+    rm rootlesskit.tar.gz
+
+RUN ARCH=$(dpkg --print-architecture) && \
+    RUNC_VERSION="v1.2.0-rc.2" && \
+    SHASUM=$(case $ARCH in \
+        amd64) echo '7bf38e9bc6c7a3feb5ec92e5e333015ccd0f360cdd1810a5a13f5ab81731cd62' ;; \
+        arm64) echo '3741f07946c54a23ee443de84506c6e397d0d350f1f16f81ae4dd2c7d39d2651' ;; \
+    esac) && \
+    curl -o "runc" -fsSL "https://github.com/opencontainers/runc/releases/download/${RUNC_VERSION}/runc.${ARCH}" && \
+    echo "$SHASUM runc" | sha256sum -c && \
+    mv runc /usr/bin/runc && \
+    chmod +x /usr/bin/runc
+
+RUN ARCH=$(dpkg --print-architecture) && \
+    CONTAINERD_BIN_VERSION="1.7.19" && \
+    SHASUM=$(case $ARCH in \
+        amd64) echo '97f75e60f0ad19d335b1d23385835df721cad4492740d50576997f2717dc3f94' ;; \
+        arm64) echo '1839e6f7cd7c62d9df3ef3deac3f404cdd5cd47bbdf8acfeb0b0f3776eb20002' ;; \
+    esac) && \
+    curl -o "containerd.tar.gz" -fsSL "https://github.com/containerd/containerd/releases/download/v${CONTAINERD_BIN_VERSION}/containerd-${CONTAINERD_BIN_VERSION}-linux-${ARCH}.tar.gz" && \
+    echo "$SHASUM containerd.tar.gz" | sha256sum -c && \
+    tar -xz -f containerd.tar.gz --to-stdout bin/containerd > /usr/bin/containerd && \
+    tar -xz -f containerd.tar.gz --to-stdout bin/containerd-shim > /usr/bin/containerd-shim && \
+    tar -xz -f containerd.tar.gz --to-stdout bin/containerd-shim-runc-v1 > /usr/bin/containerd-shim-runc-v1 && \
+    tar -xz -f containerd.tar.gz --to-stdout bin/containerd-shim-runc-v2 > /usr/bin/containerd-shim-runc-v2 && \
+    tar -xz -f containerd.tar.gz --to-stdout bin/ctr > /usr/bin/ctr && \
+    chmod +x /usr/bin/containerd /usr/bin/containerd-shim /usr/bin/containerd-shim-runc-v1 /usr/bin/containerd-shim-runc-v2 /usr/bin/ctr && \
+    rm containerd.tar.gz
+
+RUN apt-get remove -y \
+    coreutils curl ca-cerificates  apt-transport-https && \
+    apt-get autoremove -y & \
+    apt-get clean
```

Comment:
could you double-check that this `autoremove` is not removing a ton of packages and resulting in a large diff?

(I find [dive](https://github.com/wagoodman/dive) to be helpful for this sort of thing)

Reply:
haha, yeah, I definitely use dive.

It says the layer is 0 bytes, so I think we're good :P